addons/koji/common/src/main/java/org/commonjava/indy/koji/content/KojiContentManagerDecorator.java

```diff
@@ -567,6 +567,8 @@ private Group adjustTargetGroup( final RemoteRepository buildRepo, final Group s
                                                                                + buildRepo.getMetadata( NVR ) );
 
                 storeDataManager.storeArtifactStore( targetGroup, changeSummary, false, true, new EventMetadata() );
+                // Use repo service API to append the member of group instead of to override during target group update.
+                storeDataManager.addConstituentToGroup( targetGroup.getKey(), buildRepo.getKey() );
```

Comment:
If we have the addConstituentToGroup, is the above storeArtifactStore still needed ?

Reply:
@sswguo I just re-read this part, and yes, it seems the original logic is only focused on group update with a new member, so I annotated, and updated https://github.com/Commonjava/indy/pull/2507/commits/8ad742a814f80586937ec527a51ed792e9e5ff48